libs/datamodel/connectors/dml/src/native_type_instance.rs

```diff
@@ -32,3 +38,24 @@ impl NativeTypeInstance {
         serde_json::from_value(self.serialized_native_type.clone()).expect(&error_msg)
     }
 }
+
+impl TryFrom<NativeTypeInstance> for MsSqlType {
```

Comment:
The conversion from `MssqlType` to `NativeTypeInstance` does not belong in this crate. This is only the interface and must not contain implementation details.
It's also not needed. See comments in implementation of datamodel connector about `NativeTypeInstance::deserialize_native_type`.